### ISSUE TITLE
jams.trim()

### ISSUE THREAD

**Issue by justinsalamon** [marl/jams#108]
Do we have any functionality for trimming jams files? I couldn't find any, so wrote a quick example of what trimming might look like for a `pitch_midi` annotation:

``` python
def trim_jams_pitch(input_jams_path, output_jams_path, start_t, end_t):

    jams_in = jams.load(input_jams_path)

    if end_t > jams_in.file_metadata.duration:
        warnings.warn("End time provided (end_t) is greater than the duration of the "
                      "audio file.")

    onsets = []
    offsets = []
    pitches = []
    confidences = []

    # get pitch annotation (we assume there is only 1)
    pitch_ann = jams_in.annotations[0]

    # find all notes that are within the range and trim them appropriately
    for t, d, p, c in zip(pitch_ann.data.time, pitch_ann.data.duration, pitch_ann.data.value, pitch_ann.data.confidence): 
        onset = t.total_seconds()
        offset = onset + d.total_seconds()

        if onset < end_t and offset > start_t: # this should catch all overlapping notes
            t_on = max(onset, start_t)
            t_off = min(offset, end_t)
            onsets.append(t_on)
            offsets.append(t_off)
            pitches.append(p)
            confidences.append(c)

    # Create a new annotation with the trimmed notes
    trimmed_ann = jams.Annotation('pitch_midi')
    for onset, offset, pitch, conf in zip(onsets, offsets, pitches):
        trimmed_ann.append(time=onset, duration=(offset-onset), value=pitch, confidence=conf)


    # Create new jams object to store new annotation
    new_jam = jams.JAMS(annotations=[trimmed_ann])

    # Copy over file metadata, change duration field
    new_jam.file_metadata = jams_in.file_metadata
    new_jam.file_metadata.duration = end_t - start_t

    # Save new jam file
    new_jam.save(output_jams_path)
```

We could probably write a generic trimmer for any annotation namespace, or if that looks too convoluted we could have a per-namespace trimmer.

Thoughts?

**Comment by ejhumphrey** [comment]
this is implemented now, ya? close now or next minor release?

**Comment by justinsalamon** [comment]
Addressed by PR #136, closing out.

Note: discussion led to the implementation of two functions:
* trim: only keep observations within a specified time range, leaving their start times unchanged
* slice: similar to trim, but time axis reset such that the start time of the trim range is mapped to time 0 in the new sliced annotations.
